content/vi/docs/concepts/architecture/cloud-controller.md

```diff
@@ -0,0 +1,241 @@
+
+#### Route controller
+
+Route controller đóng vai trò cấu hình định tuyến trong nằm trong hệ thống cloud để các container trên các Node khác nhau trong cụm Kubernetes có thể giao tiếp với nhau. Bộ Route controller hiện chỉ đáp ứng được cho các cụm Google Compute Engine.
+
+#### Service controller
+
+Service controller lắng nghe các sự kiện như một Service được tạo, cập nhật và xóa bỏ. Dựa trên trạng thái hiện tại của các vụ trên Kubernetes, nó cấu hình các bộ tải trọng trên cloud (như ELB của AWS, Google Load Balancer, hay Oracle Cloud Infrastructure LB) nhằm phản ánh trạng thái của các Service trên Kubernetes. Mặt khác, nó đám bảo những dịch vụ phụ trợ cho những dịch vụ cân bằng tải trên cloud được cập nhật mới nhất.
+
+### 2. Kubelet
+
+Node controller bao gồm một số tính nằng phụ thuộc vào tầng cloud của Kubelet. Trước khi có bộ CCM, Kubelet đảm nhận vai trò khởi tạo một Node với thông tin chi tiết từ cloud như địa chỉ IP, phân vùng hay loại máy chủ. Với bộ CCM, vai trò này được bộ CCM đảm nhận thay cho Kubelet.
+
+Với mô hình mới này, Kubelet sẽ khởi tạo một Node nhưng không đi kèm với những thông tin từ cloud. Tuy nhiên, nó sẽ thêm vào một dấu chờ đánh dấu Node sẽ không được sắp xếp công việc (taint) cho tới khi bộ CCM khởi tạo xong Node này với những thông tin cụ thể cung cấp từ Cloud, sau đó nó sẽ xóa những dấu chờ này.
+
+## Cơ chế Plugin
+
+Bộ CCM sử dụng interface trong ngôn ngữ Go để đáp ứng việc triển khai trên bất kì hệ thống cloud nào cũng có thể cắm vào sử dụng. Cụ thể hơn, nó sử dụng CloudProvider Interface được định ở [đây](https://github.com/kubernetes/cloud-provider/blob/9b77dc1c384685cb732b3025ed5689dd597a5971/cloud.go#L42-L62).
+
+Cách triển khai của bốn thành phần trên được tô đậm ở đường dẫn trên, và một số được thực hiện như giao diện chung cho các bên cung cấp dịch vụ cloud, sẽ nằm trong chính nhân gốc của Kubernetes. Cách thực hiện cụ thể cho từng nhà cung cấp dịch vụ sẽ được xây dựng ngoài nhân gốc và triển khai các giao diện xác định trước trong core.
+
+Để biết thêm chỉ tiết, bạn có thể xem về [Cloud Controller Manager](/docs/tasks/administer-cluster/developing-cloud-controller-manager/).
+
+## Phân quyền
+
+Phần này sẽ phân nhỏ quyền hạn cần có cho các API object cung cấp bởi bộ CCM để thực hiện những hành động cần thiết.
+
+### Node controller
+
+Node controller chỉ hoạt động với các đối tượng Node. Nó yêu cầu đầy đủ quyền truy cập bao gồm get, list, create, update, patch, watch, và delete một đối tượng Node.
+
+v1/Node:
+
+- Get
+- List
+- Create
+- Update
+- Patch
+- Watch
+- Delete
+
+### Route controller
+
+Route controller lắng nghe sự kiện tạo ra các Node và cấu hình các Route tương ứng. Nó yêu cầu có quyền truy cập get tới các đối tượng Node.
+
+v1/Node:
+
+- Get
+
+### Service controller
+
+Service controller lắng nghe các sự kiện khởi tạo, cập nhật và xóa bỏ một đối tượng Service và cấu hình những điểm kết phù hợp.
```

Comment:
- `xóa bỏ một đối tượng Service` -> `xóa bỏ một Service`
- `cấu hình những điểm kết điểm kết phù hợp.` -> `cấu hình endpoints cho những Services đó một cách thích hợp`

Reply:
- `đối tượng` was intentionally used to describe the resource object in Kubernetes rather than the term Service standing alone. But if we agreed to remove it then I'll clean up all the `đối tượng` in the docs. What do you think?
- `điểm kết` to `endpoint` done!

Reply:
I think it's better if we remove `đối tượng` here.

Reply:
done anh, I will update in the next commit